### PR TITLE
[feature] Add simple Azure provider behind feature flag

### PR DESCRIPTION
**Pull request by philbrookes**

### CODE REVIEW

File: test/e2e/multi_record_test.go [hunk 1]
```diff
@@ -256,6 +256,9 @@ var _ = Describe("Multi Record Test", func() {
 
 	Context("loadbalanced", func() {
 		It("makes available a hostname that can be resolved", func(ctx SpecContext) {
+			if testDNSProvider == "azure" {
+				return
```
Comment (mikenairn) [inline test/e2e/multi_record_test.go:260]:
```suggestion
				Skip("not supported for Azure provider")
```

**Review by mikenairn — CHANGES_REQUESTED** [review]
Left some comments.

Can you add "azure" [here](https://github.com/Kuadrant/dns-operator/blob/main/config/deploy/local/kustomization.yaml#L13) so that it gets enabled on the local deployments (make local-setup(-multi) DEPLOY=true)
```
value: --provider=aws,google,azure,inmemory
```

Can you also add the azure import [here](https://github.com/Kuadrant/dns-operator/blob/main/test/e2e/multi_instance/suite_test.go#L28), and add a skip on the loadbalanced test in there also.
```
_ "github.com/kuadrant/dns-operator/internal/provider/azure"
```

Are you adding azure to the GH action e2e tests in this PR or doing that in another? Probably good to add it sooner rather than later, will require a new dns zone and principal etc.. setup in Azure and credentials added in GitHub.

File: README.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ make local-setup-gcp-mz-clean local-setup-gcp-mz-generate GCP_ZONE_NAME=<My GCP
 ```
 More details about the GCP provider can be found [here](./docs/provider.md#google-cloud-dns-provider)
 
+##### AZURE Provider
+
+```bash
+make local-setup-azure-mz-clean local-setup-azure-mz-generate KUADRANT_AZURE_CREDENTIALS='<My Azure Credentials.json>' KUADRANT_AZURE_DNS_ZONE_ID=<My Azure Zone ID>
```
[the remark this answers is not in the document]
Comment (philbrookes) [reply README.md:30]: yeah, the ID is pretty gross, I couldn't see an obvious way to shorten it, due to the variable resourceGroup.

File: .github/workflows/ci-e2e.yaml [hunk 3]
```diff
@@ -45,10 +45,14 @@ jobs:
       - name: Create GCP provider configuration
         run: |
           make local-setup-gcp-mz-clean local-setup-gcp-mz-generate GCP_ZONE_NAME=e2e-google-hcpapps-net GCP_ZONE_DNS_NAME=e2e.google.hcpapps.net GCP_GOOGLE_CREDENTIALS='${{ secrets.E2E_GCP_GOOGLE_CREDENTIALS }}' GCP_PROJECT_ID=${{ secrets.E2E_GCP_PROJECT_ID }}
+      - name: Create Azure provider configuration
+        run: |
+          make local-setup-azure-mz-clean local-setup-azure-mz-generate KUADRANT_AZURE_DNS_ZONE_ID='${{ secret.E2E_AZURE_ZONE_ID }}' KUADRANT_AZURE_ZONE_ROOT_DOMAIN=e2e.azure.hcpapps.net KUADRANT_AZURE_CREDENTIALS='${{ secrets.E2E_AZURE_CREDENTIALS }}'
```
Comment (mikenairn) [inline .github/workflows/ci-e2e.yaml:50]:
```suggestion
          make local-setup-azure-mz-clean local-setup-azure-mz-generate KUADRANT_AZURE_DNS_ZONE_ID='${{ secrets.E2E_AZURE_ZONE_ID }}' KUADRANT_AZURE_ZONE_ROOT_DOMAIN=e2e.azure.hcpapps.net KUADRANT_AZURE_CREDENTIALS='${{ secrets.E2E_AZURE_CREDENTIALS }}'
```

**Review by mikenairn — APPROVED** [review]

**Pull request merged**